wearipedia/devices/fitbit/fitbit_sense_gen.py

```diff
@@ -0,0 +1,368 @@
+
+
+def get_heart_rate(date):
+    """Generate heart rate data for a given date.
+
+    :return: dictionary of heart rate values and details
+    :rtype: dictionary
+    """
+
+    heart_rate_data = {
+        "heart_rate_day": [
+            {
+                "activities-heart": [
+                    {
+                        "dateTime": date,
+                        "value": {
+                            "customHeartRateZones": [],
+                            "heartRateZones": [
+                                {
+                                    "caloriesOut": 2877.06579,
+                                    "max": 110,
+                                    "min": 30,
+                                    "minutes": 0,
+                                    "name": "Out of Range",
+                                },
+                                {
+                                    "caloriesOut": 0,
+                                    "max": 136,
+                                    "min": 110,
+                                    "minutes": 0,
+                                    "name": "Fat Burn",
+                                },
+                                {
+                                    "caloriesOut": 0,
+                                    "max": 169,
+                                    "min": 136,
+                                    "minutes": 0,
+                                    "name": "Cardio",
+                                },
+                                {
+                                    "caloriesOut": 0,
+                                    "max": 220,
+                                    "min": 169,
+                                    "minutes": 0,
+                                    "name": "Peak",
+                                },
+                            ],
+                            "restingHeartRate": 58,
+                        },
+                    }
+                ],
+                "activities-heart-intraday": {
+                    "dataset": [],
+                    "datasetInterval": 1,
+                    "datasetType": "minute",
+                },
+            }
+        ]
+    }
+
+    the_time = time(hour=0, minute=0, second=0)
+    for i in range(1440):
+
+        max_heart_rate = [110, 136, 169, 220]
+        min_dict = {110: 30, 136: 110, 169: 136, 220: 169}
+        weights = [0.14, 0.0058, 0.0008, 0.0001]
+        max_heart = choices(max_heart_rate, weights)
+
+        inx = max_heart_rate.index(max_heart[0])
+        heart_rate_data["heart_rate_day"][0]["activities-heart"][0]["value"][
+            "heartRateZones"
+        ][inx]["minutes"] += 1
+        heart_rate_data["heart_rate_day"][0]["activities-heart"][0]["value"][
+            "heartRateZones"
+        ][inx]["caloriesOut"] += 2 * (inx + 0.5)
+
+        val = np.random.randint(min_dict[max_heart[0]], max_heart)
+
+        heart_rate_data["heart_rate_day"][0]["activities-heart-intraday"][
```

Comment:
It looks like here you are generating high frequency heart rate data randomly. I think a better way would be to use a slightly more complex generator so we do not end up with gaussian noise data. You can check out some examples of existing generators at Garmin Fenix or Polar Verity Sense!